### PR TITLE
add pipeline record Restful API

### LINKED ISSUE #174: Add pipeline record REST APIs
Need to provide the REST APIs to operate the pipeline record.

### PR DESCRIPTION
**Pull request by zz-bmj**
## Fixes   #174 

**Changes**
   
  - [x] add pipeline record Restful API

**Reviewers**

  - [x] @supereagle  please review

### CODE REVIEW

File: pkg/server/manager/pipeline.go [hunk 1]
```diff
@@ -115,15 +121,29 @@ func (m *pipelineManager) DeletePipeline(projectName string, pipelineName string
...
+		logdog.Errorf("Fail to delete the pipeline %s in the project %s as %s", pipelineName, projectName, err.Error())
 		return err
 	}
 
-	return m.dataStore.DeletePipelinesByProjectID(project.ID)
+	return nil
+}
+
+// ClearPipelinesOfProject deletes all pipelines in one project.
+func (m *pipelineManager) ClearPipelinesOfProject(projectID string) error {
```
Comment (supereagle) [inline pkg/server/manager/pipeline.go]: I have concern about this change:
* Changing "projectName" to "projectID" will affect the caller, but the update for caller is not found.
* Where to get the project id? Where to check the existence of this project?

File: pkg/server/manager/project.go [hunk 2]
```diff
@@ -35,12 +35,13 @@ type ProjectManager interface {
 
 // projectManager represents the manager for project.
 type projectManager struct {
-	dataStore       *store.DataStore
-	pipelineManager PipelineManager
+	dataStore             *store.DataStore
+	pipelineManager       PipelineManager
+	pipelineRecordManager PipelineRecordManager
```
Comment (supereagle) [inline pkg/server/manager/project.go]: `project` does not have direct relationship with `pipeline record`, so it does not need `pipelineRecordManager`.

File: store/pipelineRecord.go [hunk 3]
```diff
@@ -35,16 +35,32 @@ func (d *DataStore) CreatePipelineRecord(pipelineRecord *api.PipelineRecord) (*a
 	return pipelineRecord, nil
 }
 
-// FindPipelineRecordsByPipelineID finds the pipeline record by pipelineID.
-func (d *DataStore) FindPipelineRecordsByPipelineID(pipelineID string) ([]api.PipelineRecord, error) {
+// FindPipelineRecordsByPipelineID finds the pipeline records by pipelineID.
+func (d *DataStore) FindPipelineRecordsByPipelineID(pipelineID string, queryParams api.QueryParams) ([]api.PipelineRecord, int, error) {
+	pipelineRecords := []api.PipelineRecord{}
 	query := bson.M{"pipelineID": pipelineID}
```
Comment (supereagle) [inline store/pipelineRecord.go]: `pipelineID` has been changed to `pipelineId` in PR #167. More test will be appreciated.

File: store/pipelineRecord.go [hunk 4]
```diff
@@ -59,13 +75,17 @@ func (d *DataStore) FindPipelineRecordByID(pipelineRecordID string) (*api.Pipeli
...
 
-// DeletePipelineRecord deletes the pipeline record by id.
-func (d *DataStore) DeletePipelineRecord(pipelineRecordID string) error {
+// DeletePipelineRecordByID deletes the pipeline record by id.
+func (d *DataStore) DeletePipelineRecordByID(pipelineRecordID string) error {
 	return d.pipelineRecordCollection.RemoveId(pipelineRecordID)
 }
+
+// DeletePipelineRecordByPipelineID deletes all the pipeline records of one pipeline by pipeline id.
+func (d *DataStore) DeletePipelineRecordByPipelineID(pipelineID string) error {
```
Comment (supereagle) [inline store/pipelineRecord.go]: s/DeletePipelineRecordByPipelineID/DeletePipelineRecordsByPipelineID/g

File: pkg/server/manager/project.go (hunk 2 above)
Comment (zz-bmj) [reply pkg/server/manager/project.go]: Oh, I forgot to delete it, I had fixed it.

File: pkg/server/manager/pipeline.go (hunk 1 above)
Comment (zz-bmj) [reply pkg/server/manager/pipeline.go]: now, only the func DeleteProject in pkg/server/manager/project.go call it. and the DeleteProject have checked the existence of this project, I think we don't need to check it twice. or do you have any better suggestion?

File: store/pipelineRecord.go (hunk 3 above)
Comment (zz-bmj) [reply store/pipelineRecord.go]: fixed

File: store/pipelineRecord.go (hunk 4 above)
Comment (zz-bmj) [reply store/pipelineRecord.go]: fixed

File: pkg/server/manager/pipeline.go (hunk 1 above)
Comment (supereagle) [reply pkg/server/manager/pipeline.go]: Use `projectID` is OK.

**Comment by supereagle** [comment]
LGTM

**Pull request merged**